app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/AddCoInsuredBottomSheetContent.kt

```diff
@@ -226,6 +237,43 @@ private fun FetchFromSsnFields(
   }
 }
 
+private class PersonalNumberVisualTransformation(
+  private val mask: String,
+  private val maskColor: Color,
+) : VisualTransformation {
+  override fun filter(text: AnnotatedString): TransformedText {
+    val trimmed = if (text.text.length >= 12) text.text.substring(0..11) else text.text
+
+    val annotatedString = buildAnnotatedString {
+      for (i in trimmed.indices) {
+        append(trimmed[i])
+        if (i == 7) {
+          append("-")
+        }
+      }
+      withStyle(SpanStyle(color = maskColor)) {
+        append(mask.takeLast(mask.length - length))
+      }
+      toAnnotatedString()
```

Comment:
You can remove this here too, since internally the function calls it automatically
```kotlin
inline fun buildAnnotatedString(builder: (Builder).() -> Unit): AnnotatedString =
    Builder().apply(builder).toAnnotatedString()
```
It was part of why I wanted to use this dsl too, since it does this part for you and does not leave you room to make a mistake of not calling it otherwise